sdk/tests/invariant.test.ts

```diff
@@ -245,13 +252,162 @@ describe('invariant', async () => {
     const pools = await invariant.getPools(account)
     assert.deepEqual(pools.length, 1)
   })
+
+  describe('positions', async () => {
+    beforeEach(async () => {
+      await invariant.addFeeTier(account, feeTier)
+
+      await invariant.createPool(
+        account,
+        token0.contract.address.toString(),
+        token1.contract.address.toString(),
+        feeTier,
+        { v: 1000000000000000000000000n },
+        0n
+      )
+
+      await token0.approve(account, invariant.contract.address.toString(), 10000000000000n)
+      await token1.approve(account, invariant.contract.address.toString(), 10000000000000n)
+
+      const poolKey = _newPoolKey(
+        token0.contract.address.toString(),
+        token1.contract.address.toString(),
+        feeTier
+      )
+
+      const result = await invariant.createPosition(
+        account,
+        poolKey,
+        -10n,
+        10n,
+        { v: 10000000000000n },
+        { v: 1000000000000000000000000n },
+        { v: 1000000000000000000000000n }
+      )
+
+      assert.deepEqual((result.events[0] as any).address, account.address.toString())
```

Comment:
Is there a result that returns a contract when you call the entry point always the last event in the array? It seems it should be the last. If so, we can utilize this fact and return strongly typed events